ntbs-service/Helpers/NotificationHelper.cs

```diff
@@ -15,7 +15,12 @@ public static void SetShouldValidateFull(Notification notification)
                     ownedModel.GetType().GetProperty("ShouldValidateFull").SetValue(ownedModel, true);
                 }
             }
-            notification.NotificationSites?.ForEach(x => x.ShouldValidateFull = notification.ShouldValidateFull);
+            notification.NotificationSites?.ForEach(ntfSite => {
+                if (ntfSite != null)
+                {
+                    ntfSite.ShouldValidateFull = notification.ShouldValidateFull;
+                }
+            });

```

Comment:
It's a little worrying that we need to account for that - why would we ever have a `null` site?